cmd/util/ledger/migrations/account_migration.go

```diff
@@ -0,0 +1,94 @@
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/meter"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+)
+
+func MigrateAccountUsage(payloads []ledger.Payload, pathFinderVersion uint8) ([]ledger.Payload, []ledger.Path, error) {
+	return MigrateByAccount(AccountUsageMigrator, payloads, pathFinderVersion)
+}
+
+func KeyToStorageInteractionKey(key ledger.Key) (meter.StorageInteractionKey, error) {
+	id, err := KeyToRegisterID(key)
+	if err != nil {
+		return meter.StorageInteractionKey{}, err
+	}
+	return meter.StorageInteractionKey{
+		Owner: id.Owner,
+		Key:   id.Key,
+	}, nil
+}
+
+func payloadSize(key ledger.Key, payload ledger.Payload) (uint64, error) {
+	ik, err := KeyToStorageInteractionKey(key)
+	if err != nil {
+		return 0, err
+	}
+
+	return meter.GetStorageKeyValueSizeForTesting(ik, payload.Value()), nil
+}
+
+func isAccountKey(key ledger.Key) bool {
+	return string(key.KeyParts[1].Value) == state.AccountStatusKey
+}
+
+// AccountUsageMigrator iterate through each payload, and calculate the storage usage
+// and update the accoutns status with the updated storage usage
+func AccountUsageMigrator(account string, payloads []ledger.Payload, pathFinderVersion uint8) ([]ledger.Payload, error) {
+	var status *environment.AccountStatus
+	var statusIndex int
+	totalSize := uint64(0)
+	for i, payload := range payloads {
+		key, err := payload.Key()
+		if err != nil {
+			return nil, err
+		}
+		if isAccountKey(key) {
+			statusIndex = i
+			status, err = environment.AccountStatusFromBytes(payload.Value())
+			if err != nil {
+				return nil, fmt.Errorf("could not parse account status: %w", err)
+			}
+
+		}
+
+		size, err := payloadSize(key, payload)
+		if err != nil {
+			return nil, err
+		}
+		totalSize += size
+	}
+
+	if status == nil {
+		return nil, fmt.Errorf("could not find account status for account %v", account)
+	}
+
+	// update storage used
+	status.SetStorageUsed(totalSize)
```

Comment:
In this example, the totalSize is supposed to be equal to the cached value from `status.StorageUsed()`. 

But in real migration, some payload will be removed, and some payload might be added, in that case, the StorageUsed might be different.